src/Google/Http/Batch.php

```diff
@@ -61,26 +61,38 @@ public function execute()
   {
     $body = '';
     $classes = array();
+    $batchHttpTemplate = <<<EOF
+--%s
+Content-Type: application/http
+Content-Transfer-Encoding: binary
+MIME-Version: 1.0
+Content-ID: %s
+
+%s%s
+%s
+
+
+EOF;
 
     /** @var Google_Http_Request $req */
     foreach ($this->requests as $key => $request) {
-      $request->addHeaders(
-          [
-            'Content-Type' => 'application/http',
-            'Content-Transfer-Encoding' => 'binary',
-            'MIME-Version' => '1.0',
-            'Content-ID' => $key,
-          ]
-      );
-      $body .= "--{$this->boundary}";
-      $body .= Request::getHeadersAsString($request) . "\n\n";
-      $body .= sprintf(
+      $firstLine = sprintf(
           '%s %s HTTP/%s',
           $request->getMethod(),
           $request->getResource(),
           $request->getProtocolVersion()
       );
-      $body .= "\n\n";
+
+      $content = (string) $request->getBody();
+
+      $body .= sprintf(
+          $batchHttpTemplate,
+          $this->boundary,
+          $key,
+          $firstLine,
+          Request::getHeadersAsString($request),
```

Comment:
Does `Request::getHeadersAsString()` have a newline at the beginning? Otherwise it will be in the same line with the `$firstLine` as far as I read the code.